src/main/java/Duke.java

```diff
@@ -1,10 +1,123 @@
+import java.util.Iterator;
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
     public static void main(String[] args) {
+        initDuke();
+        String lineLogo = "-----------------------------";
+        Scanner in = new Scanner(System.in);
+        String line = " ";
+        String command = " ";
+        String commandSubject = " ";
+        String adverb = " ";
+        String timeline = " ";
+        ArrayList<Task> taskArrayList = new ArrayList<>();
+        Task currentTask;
+        String taskIncomplete = Character.toString((char) 0x274C);
+        String taskComplete = Character.toString((char) 0x2713);
+        int endPos = 0;
+        int slashPos = 0;
+
+        while(true){
+            line = in.nextLine();
+
+            if(line.equals("bye")){
+                break;
+            }
+
+            if(line.contains(" ")){
+                endPos = line.indexOf(" ");
+                command = line.substring(0, endPos);
+                commandSubject = line.substring(endPos + 1);
+                if(line.contains("/")){
+                    slashPos = commandSubject.indexOf("/");
+                    timeline = commandSubject.substring(slashPos + 3);
+                    commandSubject = commandSubject.substring(0, slashPos - 1);
+                }
+            }
+            else{
+                command = line;
+            }
+
+
+            switch(command){
+            case "todo":
+                taskArrayList.add(new Task('T', false, commandSubject, null));
+                displayToDo(lineLogo, commandSubject, taskIncomplete, Task.getTotalTasks(), Task.getTaskType());
+                break;
+
+            case "deadline":
+                taskArrayList.add(new Task('D', false, commandSubject, timeline));
+                displayToDoWithTime(lineLogo, commandSubject, taskIncomplete, Task.getTotalTasks(),
+                        Task.getTaskType(), timeline);
+                break;
+
+            case "event":
+                taskArrayList.add(new Task('E', false, commandSubject, timeline));
+                displayToDoWithTime(lineLogo, commandSubject, taskIncomplete, Task.getTotalTasks(), Task.getTaskType(), timeline);
+                break;
+
+            case "done":
+                int taskNum = Integer.parseInt(commandSubject);
+                taskNum--;
+                currentTask = taskArrayList.get(taskNum);
+                currentTask.setIsTaskComplete(true);
+                System.out.println(lineLogo);
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println("   ["  + currentTask.getTaskType() + "] [" + taskComplete + "] "+
+                        currentTask.getTaskName());
+                System.out.println(lineLogo);
+                break;
+
+            case "list":
+                break;
```

Comment:
I feel that this part could be broken into different function/class for better abstraction